src/streaming/streamingclient.py

```diff
@@ -0,0 +1,178 @@
+"""StreamingClient tool used for streaming services"""
+from AudioConfig import AudioConfig
+import websocket
+import io
+import threading
+import six
+import json
+
+def on_error(error):
+    pass
+
+def on_close(code, reason):
+    pass
+
+def on_open():
+    pass
+
+class RevAiStreamingClient():
+    def __init__(self, 
+                access_token, 
+                config, 
+                version = 'v1alpha',  
+                on_error = on_error, 
+                on_close = on_close, 
+                on_open = on_open):
+        """Constructor for Streaming Client
+
+        :param access_token: access token which authorizes all requests and links them to your
+                             account. Generated on the settings page of your account dashboard
+                             on Rev.ai.
+        :param config: an AudioConfig object containing audio information. 
+                       See AudioConfig.py for more information
+        :param version (optional): version of the streaming speech to text as string
+        :param on_message (optional): function to be called when recieving a message from the server
+        :param on_error (optional): function to be called when recieving an error from the server
+        :param on_close (optional): function to be called when the websocket closes
+        :param on_open (optional): function to be called when the websocket and thread starts successfully
+        """
+        if not access_token:
+            raise ValueError('access_token must be provided')
+
+        if not config:
+            raise ValueError('config must be provided')
+
+
+        self.access_token = access_token
+        self.config = config
+        self.base_url = \
+                'wss://api.rev.ai/speechtotext/{}/stream'.format(version)
+        self.on_error = on_error
+        self.on_close = on_close
+        self.on_open = on_open
+        self.thread = SpeechThread
+        self.client = websocket.WebSocket(enable_multithread = True, 
+            on_error = self.on_error, 
+            on_close = self.on_close, 
+            on_open = self.on_open
+            )
+
+    def start(self):
```

Comment:
This method should return the responses() generator and take a generator as input.

My reasoning for this is that I don't believe a customer would want to start the streaming connection and not begin sending and receiving from the server. By doing this we can combine 3 method calls into one. 

Before
```
client.start()
client.send_data_as_gen(input)
for i in client.responses(): ...
```
After
```
responses = client.start(input)
for i in responses: ...
```